go/stream/message_test.go

```diff
@@ -110,6 +110,76 @@ func TestMessage(t *testing.T) {
 		require.Equal(t, input, propertiesOut)
 	})
 
+	t.Run("properties to/from: pulsar with bot fields", func(t *testing.T) {
+		input := map[string]string{
+			"requestType":       "requestType",
+			"routingKey":        "routingKey",
+			"workspaceID":       "workspaceID",
+			"userID":            "userID",
+			"sourceID":          "sourceID",
+			"destinationID":     "destinationID",
+			"requestIP":         "10.29.13.20",
+			"receivedAt":        time.Date(2024, 8, 1, 0o2, 30, 50, 200, time.UTC).Format(time.RFC3339Nano),
+			"sourceJobRunID":    "sourceJobRunID",
+			"sourceTaskRunID":   "sourceTaskRunID",
+			"traceID":           "traceID",
+			"compression":       "some-serialized-compression-settings",
+			"encryption":        "some-serialized-encryption-settings",
+			"encryptionKeyID":   "encryptionKeyID",
+			"isBot":             "true",
```

Comment:
Can we have a test where we start from a map that has `isBot:false`, then you assert on the expected properties such as:
* botName must be an empty string
* botURL must be an empty string
* botUnknownBrowser must be an empty string

Then we convert the properties back into the map and `botName`, `botURL` and `botUnknownBrowser` should not be there in the map at all (I'm not saying empty, I'm saying the map should not have those keys).